renku_notebooks/api/classes/server.py

```diff
@@ -975,24 +975,38 @@ def stop(self, forced=False):
         else:
             return status
 
-    def get_logs(self, max_log_lines=0, container_name="jupyter-server"):
-        """Get the logs of the k8s pod that runs the user server."""
+    def get_logs(self, max_log_lines=0):
+        """Get the logs of all containers in the server pod."""
         js = self.js
         if js is None:
             return None
+        output = {}
         pod_name = js["status"]["mainPod"]["name"]
-        if max_log_lines == 0:
-            logs = self._k8s_client.read_namespaced_pod_log(
-                pod_name, self._k8s_namespace, container=container_name
-            )
+        all_containers = js["status"]["mainPod"]["status"].get(
+            "containerStatuses", []
+        ) + js["status"]["mainPod"]["status"].get("initContainerStatuses", [])
+        for container in all_containers:
+            container_name = container["name"]
+            try:
+                logs = self._k8s_client.read_namespaced_pod_log(
+                    pod_name,
+                    self._k8s_namespace,
+                    container=container_name,
+                    tail_lines=max_log_lines if max_log_lines > 0 else None,
+                    timestamps=True,
+                )
+            except ApiException as err:
+                if err.status in [400, 404]:
+                    continue  # container does not exist or is not ready yet
+                else:
+                    raise err
```

Comment:
```suggestion
                    raise
```

That way the original stack trace is preserved.

Even better is `raise DedicatedExceptionThatLogCantBeReadForSomeReasonException() from err`, then you get two exceptions/stack traces, the original one followed by this follow up error (naming left up to the reader) that could add additional detail like `Couldn't read log from container XYZ`.

Reply:
@Panaetius I will do this `raise DedicatedExceptionThatLogCantBeReadForSomeReasonException() from err` or something very similar in [another PR](https://github.com/SwissDataScienceCenter/renku-notebooks/pull/849) that is already open that is supposed to cleanup and standardize all the errors in the notebook service.